patches/api/0439-PreSpawnerSearchEvent.patch

```diff
@@ -0,0 +1,121 @@
++
++/**
++ * Represents an event that is fired before a mob spawner attempts to search
++ * for nearby players to activate. This event occurs before any of the spawner logic,
++ * allowing cancellation to prevent the spawner from searching and spawning any mobs.
++ */
++public class PreSpawnerSearchEvent extends Event implements Cancellable {
++    @NotNull private final Location spawnerLocation;
++    @NotNull private final Collection<Player> activatingPlayers;
++    private int backoffTicks;
++    private boolean cancelled = false;
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    @ApiStatus.Internal
++    public PreSpawnerSearchEvent(@NotNull Location spawnerLocation, @NotNull Collection<Player> activatingPlayers, int backoffTicks) {
++        this.spawnerLocation = Preconditions.checkNotNull(spawnerLocation, "Spawner Location can not be null");
++        this.activatingPlayers = Preconditions.checkNotNull(activatingPlayers, "There must be a activating player");
++        this.backoffTicks = backoffTicks;
++    }
++
++    /**
++     * Retrieves the location of the mob spawner associated with the event
++     * @return The location of the spawner
++     */
++    @NotNull
++    public Location getSpawnerLocation() {
++        return spawnerLocation;
++    }
++
++    /**
++     * Retrieves an immutable collection of players who are close enough to activate the spawner
++     * @return A collection of activating players
++     */
++    @NotNull
++    public Collection<Player> getActivatingPlayers() {
++        return Collections.unmodifiableCollection(activatingPlayers);
++    }
++
++    /**
++     * Retrieves the number of backoff ticks before the mob spawner searches for players again
++     *
++     * <p>Backoff ticks determine the duration the spawner pauses its search for players. A higher value
++     * means a longer delay before the spawner attempts to search for activating players again. It's important
++     * to note that negative backoff ticks do not make sense and will be corrected to 0 automatically.</p>
++     *
++     * <p>While backoff ticks are active, the spawner suspends its search for players and will not trigger
++     *  the spawning of monsters. This effectively postpones the activation of the spawner until the backoff
++     *  ticks expire.</p>
++     *
++     * @return The amount of backoff ticks before the spawner searches for players again
++     */
++    public int getBackoffTicks() {
++        return backoffTicks;
++    }
++
++    /**
++     * Sets the number of backoff ticks before the mob spawner searches for players again.
++     *
++     * @see PreSpawnerSearchEvent#getBackoffTicks() on an explanation of how backoff ticks function
++     *
++     * @param ticks sets the amount of ticks before the spawner searches again for the player
++     * @throws IllegalArgumentException If the provided backoff ticks value is negative.
++     */
++    public void setBackoffTicks(int ticks) {
```

Comment:
`@Range(from = 0, to = Integer.MAX_VALUE)` on the parameter.

Reply:
Annotations aren't allowed for Void functions.

Reply:
The annotation should go on the int parameter but i wonder if zero really makes sense?